src/main/resources/scripts/cleanup/non_prod_primary_and_replica_data_cleanup.sql

```diff
@@ -0,0 +1,82 @@
+--
+-- RWA-2482 cleanup old tasks from primary task database
+--
+
+CREATE OR REPLACE FUNCTION task_cleanup_primary()
+RETURNS VOID
+LANGUAGE plpgsql
+AS $function$
+DECLARE
+    id text;
+BEGIN
+    -- Loop through the rows of the tasks table
+    FOR id IN select task_id from cft_task_db.tasks WHERE EXTRACT(YEAR FROM created) in (2021) LOOP
+
+        -- delete from sensitive_task_event_logs
+        delete from cft_task_db.sensitive_task_event_logs where task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.sensitive_task_event_logs where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in sensitive_task_event_logs table', id;
+        END IF;
+
+        -- delete from task_roles
+        delete from cft_task_db.task_roles where task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.task_roles where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in task_roles table', id;
+        END IF;
```

Comment:
If you can't delete the role then the task delete will fail so probably want to abort that task_id otherwise you'll get an exception which you will have to handle.  Otherwise the whole thing would stop.

Reply:
Made the necessary changes. If any issue occurs while deleting in the child tables, i am skipping that task ID deletion in the tasks table.

Reply:
👍